ToDoApp/ToDoAppTests/ToDoServiceTests.swift

```diff
@@ -29,6 +30,8 @@ final class ToDoServiceTests: XCTestCase {
             capturedValue = value
             // Mock repository behaviour returns nil error
             return nil
+        }, read: {
+            return .failure(RepositoryError.createOrUpdateError)
```

Comment:
It would be good execute XCTFail() since success test case doesn't expect this method to be invoked

Reply:
I think this will generate error because it's supposed to return Result<[TaskModel], RepositoryError> so when substitute with XCTFail() the return does not match the required return type

Reply:
It can be done just before return, do xcode would throw test failure if unexpected code path would be executed

Reply:
Hi, it's been such a long time. But I have fixed it. And when running it, it said build failed, but then still passed the test.